### DISCUSSION TITLE
能开发一个后台吗

### DISCUSSION THREAD

**Discussion by sysfox** [vuepress/core#1218] [Ideas]
Hexo有后台项目Qexo、HPP，在我看来，Vuepress对于一个博客人来说一个后台也是必要的（可以独立开个repo研发，供有需求的人使用）
开发的难度可能较大，但毕竟Vuepress2是一个全新的项目，可以考虑一下

**Comment by Mister-Hope** [comment]
<img width="425" alt="image" src="https://user-images.githubusercontent.com/33315834/209900993-538fd31e-e78b-46b0-b12a-a7d4d5ebf2e0.png">

西餐厅能卖豆浆油条么？能，也不能。

**Comment by sysfox** [comment]
不一定非要卖豆浆油条，也可以提供刀叉
可以像Qexo那样，调用git仓库的api编辑markdown文件并提交
而非从数据库拉取文章形成动态框架
@Mister-Hope

**Reply by Mister-Hope** [reply]
官网的左下角就有个 edit this page 供你跳转到对应的平台直接在 web editor 中编辑。

如果你需要的是一个网页版的编辑和预览功能，那对不起:

1. 团队应该没人赞同把 markdown-it 解析器和所有 markdown-it 插件，以及一个完整的 vue 解析器/打包器塞到客户端。
2. 现有插件 API 不存在响应客户端修改的可能，如果你需要触发对应的hook，你需要把整个源文件加所有依赖都运行到虚拟环境里并提供 node api 模拟。这里你可以参考 stackblitz.com

总上，不能。

按照比喻，我给你刀叉和面粉豆子你就能自己做油条豆浆了？

**Reply by sysfox** [reply]
1.我没说全塞** 一个 **客户端里啊(哭, 不能独立开出来一个项目吗
2.我说了是在web端对git仓库的api发起请求
(试图进行最后的尝试)

**Reply by sysfox** [reply]
况且,油条豆浆是店家制做的,我们作为用户只管消费使用

**Reply by Mister-Hope** [reply]
不是，你先看看这东西跟你想的有多大差距:

![image](https://user-images.githubusercontent.com/33315834/209948280-be541c7f-c172-4049-83d4-d772fe368b98.png)

如果你想魔改的话 ，欢迎自己起项目，我们没什么兴趣就是了。你的标题叫“能开发一个后台么”，可能但团队这俩人大概都没兴趣。

**Reply by Mister-Hope** [reply]
就我觉得我在回复中已经充分表达了这种想法的不现实和不感兴趣了。

**Comment by sysfox** [comment]
好吧,抱歉浪费了您这么多时间,但同时也祝Vuepress能在以后发展得更好

**Comment by DreamOfIce** [comment]
说实话，Qexo本身和Hexo官方没有任何关系，你完全可以自己魔改Qexo或从头开发一个后台
